src/client/containers/CartPage/CartPage.Container.js

```diff
@@ -20,17 +21,45 @@ const CartPageContainer = ({ isAuthenticated }) => {
   const [cartItem, setCartItem] = React.useState([]);
   const [isLoading, setIsLoading] = React.useState(true);
   const [total, setTotal] = React.useState(0);
-
   const [user, setUser] = React.useState({});
   const [favorite, setFavorite] = React.useState([]);
   const { auth } = useFirebase();
-
   const { shoppingCart, changeProductQuantity } = useShoppingCartContext();
   const history = useHistory();
-
   const userId = (isAuthenticated && auth.currentUser.uid) || '';
-
   const userInfo = useFetchApi(`users/${userId}`);
+
+  const handleReviewOrder = () => {
+    fetch(`/api/orders`, {
+      method: 'POST',
+      headers: {
+        Accept: 'application/json, text/plain, */*',
+        'Content-Type': 'application/json',
+      },
+      body: JSON.stringify({
+        user_id: `${userId}`,
+        items: cartItem,
+      }),
+    }).then((response) => {
+      if (response.ok) {
+        console.log('Success: Order created successfully');
+      } else {
+        throw new Error(response.status);
+      }
+    });
+    fetch(`api/orders/user/${userId}`, {
```

Comment:
You can modify your router to return order id, so you donn`t have to make this fetch. You are recieving it in it actualy.

Reply:
```
const storeNewOrder = async (data) => {
  let orderId;
  await knex('orders')
    .insert({ user_id: data.user_id })
    .returning('id')
    .then((id) => {
      orderId = id;
      data.items.forEach(async (item) => {
        await knex('order_items').insert({
          order_id: id[0],
          product_id: item.product_id,
          quantity: item.quantity,
        });
      });
    });
    return orderId;
};
```